validator/val_sigcrypt.c

```diff
@@ -1677,6 +1678,9 @@ dnskey_verify_rrset_sig(struct regional* region, sldns_buffer* buf,
 	/* verify */
 	sec = verify_canonrrset(buf, (int)sig[2+2],
 		sigblock, sigblock_len, key, keylen, reason);
+
+	/* count validation operation */
+	qstate->env->mesh->val_ops++;
```

Comment:
The mesh is a null pointer, in some cases. This fix then does not crash but also does not count that validation operation. That would not count offline ZONEMD validation operations, when the server is started and reads authority zones. I think this is why the ci run is failed.
```suggestion
	if(qstate->env->mesh)
		qstate->env->mesh->val_ops++;
```

Reply:
Ah good point, I missed that.

Reply:
Hmm, this doesn't solve it:

```
0x000055555559e22d in dnskey_verify_rrset_sig (region=<optimized out>, buf=<optimized out>, ve=0x7fffffffdaa0, 
    now=<optimized out>, rrset=0x55555582e3c0, dnskey=<optimized out>, dnskey_idx=0, sig_idx=0, sortree=0x7fffffffd828, 
    buf_canon=0x7fffffffd824, reason=0x7fffffffda18, reason_bogus=0x0, section=LDNS_SECTION_ANSWER, qstate=0x0)
    at validator/val_sigcrypt.c:1683
1683		if(qstate->env->mesh)
```

Apparently `qstate` can also be `NULL`, so I guess the code should be something like:

```
if (qstate && qstate->env && qstate->env->mesh) [...]
```

Reply:
Pushed a fix.